### PR TITLE
DRIVERS-2585 Unify Secrets Handling in Drivers Tools

### PR DESCRIPTION
**Pull request by blink1073**
[Passing Python Patch Build - master](https://spruce.mongodb.com/version/65c3ab3d9ccd4e62dd817a66/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

[Passing Go Patch - v1](https://spruce.mongodb.com/version/65c3a7c932f41714fea8c0e7/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

[Python patch build utilizing these changes](https://spruce.mongodb.com/version/65c504273e8e86952df03687/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

[Python PR](https://github.com/mongodb/mongo-python-driver/pull/1517)

[Go Driver PR](https://github.com/mongodb/mongo-go-driver/pull/1564)

### CODE REVIEW

File: .evergreen/auth_oidc/README.md [hunk 1]
```diff
@@ -22,13 +22,14 @@ OIDC_ISSUER_2_URI       # The issuer URI for mock IdP 2
...
-See the 
 The `oidc_get_tokens.sh` script will automatically fetch the credentials from the `drivers/oidc` vault.
+See [Secrets Handling](../secrets_handling/README.md) for details on how the script accesses the vault.
+Add `secrets-export.sh` to your `.gitignore` to prevent checking in credentials in your repo.
 
 ### Usage
 
-Use the `oidc_get_tokens.sh` script to create a set of OIDC tokens in a temporary, including
-`test_user1` and `test_user1_expires`.  The temp file location is exported as `OIDC_TOKEN_DIR`
+Use the `oidc_get_tokens.sh` script to create a set of OIDC tokens in a temporary director, including
```
Comment (NoahStapp) [inline .evergreen/auth_oidc/README.md:31]:
```suggestion
Use the `oidc_get_tokens.sh` script to create a set of OIDC tokens in a temporary directory, including
```

File: .evergreen/auth_oidc/README.md [hunk 2]
```diff
@@ -22,13 +22,14 @@ OIDC_ISSUER_2_URI       # The issuer URI for mock IdP 2
...
+`test_user1` and `test_user1_expires`.  The temp file location is exported as `OIDC_TOKEN_DIR`.
 
 ```bash
```
Comment (NoahStapp) [inline .evergreen/auth_oidc/README.md:34]: unfinished ```?

File: .evergreen/auth_oidc/README.md [hunk 3]
```diff
@@ -44,8 +45,6 @@ to each local development and testing.
 local docker container running `mongo-orchestration` with OIDC enabled.
```
Comment (NoahStapp) [inline .evergreen/auth_oidc/README.md:45]: What is this paragraph tied to?

File: .evergreen/secrets_handling/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+If using a Linux host on Evergreen, the shorthand version of the script can be used to get the credentials, e.g.
+
+```yaml
+- command: subprocess.exec
+    params:
+    working_dir: src
+    binary: bash
+    args: |
+        ${DRIVERS_TOOLS}/.evergreen/atlas/setup-secrets.sh
```
Comment (NoahStapp) [inline .evergreen/secrets_handling/README.md:24]: Should this be `secrets_handling/setup-secrets.sh`?

**Review by NoahStapp — CHANGES_REQUESTED** [review]
Incredible work, Steve. This will be a massive help! Just a few minor clarifications.

File: .evergreen/atlas/setup-atlas-cluster.sh [hunk 5]
```diff
@@ -18,6 +18,23 @@ set -o errexit  # Exit the script with error if any of the commands fail
...
+# Set up the common variables.
+CURRENT_DIR=$(pwd)
+SCRIPT_DIR=$(dirname ${BASH_SOURCE[0]})
+. $SCRIPT_DIR/../handle-paths.sh
+pushd $SCRIPT_DIR
+
+# Load the secrets file if it exists.
+if [ -f ./secrets-export.sh ]; then
+  echo "Sourcing secrets"
+  source ./secrets-export.sh
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/atlas/setup-atlas-cluster.sh:30]: The secrets export files have `+x` at the start.

File: .evergreen/auth_oidc/README.md (hunk 2 above)
Comment (blink1073) [reply .evergreen/auth_oidc/README.md:34]: I don't follow, this block looks correct to me.

File: .evergreen/secrets_handling/README.md (hunk 4 above)
Comment (blink1073) [reply .evergreen/secrets_handling/README.md:24]: Added more explanation.

File: .evergreen/auth_oidc/README.md (hunk 2 above)
Comment (alcaeus) [reply .evergreen/auth_oidc/README.md:34]: @NoahStapp the code block is correctly closed in line 37, although this is hidden by the collapsed diff.

**Comment by blink1073** [comment]
> Is there a smart way to unset all credentials for an evergreen build to make sure all necessary credentials are fetched from the secrets vault? The only options I can think of are duplicating the project or overriding the project configuration, which would break all other builds as well.

I don't think so, I view these as best-effort until we actually remove the credentials in the project and see what breaks.   I audited inline expansions and the use of `add_expansions_to_env`.

File: .evergreen/auth_aws/README.md [hunk 6]
```diff
@@ -10,13 +10,12 @@ For all testing variants except for ECS, the general test flow is:
 - Set up the required secrets for the the test variant:
 
 ```bash
-cd $DRIVERS_TOOLS/.evergreen/auth_aws
-# Fetch the secrets from the vault and write to a local `secrets-export.sh` file.
-bash setup_secrets.sh drivers/aws_auth
+# Fetch the secrets from the vault and write to a `secrets-export.sh` file in this directory.
+bash $DRIVERS_TOOLS/.evergreen/auth_aws/setup-secrets.sh
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_aws/README.md:14]: Updated

File: .evergreen/auth_oidc/azure/README.md [hunk 7]
```diff
@@ -9,8 +9,8 @@ and then tearing down the VM and its resources.
 There are a set of scripts that facilitate these operations.
 They build on top of the scripts used in `csfle/azurekms`.
 
-See https://wiki.corp.mongodb.com/display/DRIVERS/Using+AWS+Secrets+Manager+to+Store+Testing+Secrets for more background
-on how the secrets are managed for these tests.  These secrets are used to log in to Azure, and the 
+See [Secrets Handling](../secrets_handling/README.md) for details on how the script accesses the `drivers/azureoidc` vault.  
+These secrets are used to log in to Azure, and the 
 rest of the secrets are fetched from the "OIDC-Key-Vault" in our Drivers Azure Subscription (https://portal.azure.com/#home).
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/azure/README.md:14]: Fixed

File: .evergreen/aws_lambda/README.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# AWS Lamba Testing
+
+See the [FaaS Automated Testing specification](https://github.com/mongodb/specifications/blob/master/source/faas-automated-testing/faas-automated-testing.rst) for details.
+
+See [Atlas README](../atlas/README.md) for details on how to access the secrets
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/aws_lambda/README.md:5]: Done

**Review by alcaeus — APPROVED** [review]

**Comment by blink1073** [comment]
I added some shim files to the old csfle script names that are only used by python and go, to be removed once the linked PRs are merged.

**Pull request merged**